onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -281,9 +281,21 @@ class PlannerImpl {
     return elt_type->Size();
   }
 
-  static bool SameSize(const TensorShapeProto& shape1, const DataType& ptype1, const TensorShapeProto& shape2,
-                       const DataType& ptype2) {
-    return (GetElementSize(ptype1) == GetElementSize(ptype2)) && SameShape(shape1, shape2);
+  static bool SameSize(const TensorShapeProto& shape1, const onnxruntime::NodeArg& arg1, const TensorShapeProto& shape2, const onnxruntime::NodeArg& arg2) {
+    const auto& ptype1 = arg1.Type();
+    const auto& ptype2 = arg2.Type();
+    auto type1_size = GetElementSize(ptype1);
```

Comment:
Would it be easier to work with TypeProto from NodeArg::TypeAsProto instead?

Reply:
The logic for GetElementSize existed before. I separated it out on different lines for easier debugging.

Reply:
I meant for working out if it's a string or not. e.g.
    bool is_type1_string = arg1.TypeAsProto()->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType_STRING;


---
In reply to: [346083427](https://github.com/microsoft/onnxruntime/pull/2393#discussion_r346083427) [](ancestors = 346083427)

Reply:
Ah, sure.